libraries/AP_InertialSensor/AP_InertialSensor_tempcal.cpp

```diff
@@ -0,0 +1,219 @@
+
+    // @Param: GYR1_X
+    // @DisplayName: Gyroscope 1st order temperature coefficient X axis
+    // @Description: This is the 1st order temperature coefficient from a temperature calibration
+    // @User: Advanced
+    // @Calibration: 1
+
+    // @Param: GYR1_Y
+    // @DisplayName: Gyroscope 1st order temperature coefficient Y axis
+    // @Description: This is the 1st order temperature coefficient from a temperature calibration
+    // @User: Advanced
+    // @Calibration: 1
+
+    // @Param: GYR1_Z
+    // @DisplayName: Gyroscope 1st order temperature coefficient Z axis
+    // @Description: This is the 1st order temperature coefficient from a temperature calibration
+    // @User: Advanced
+    // @Calibration: 1
+    
+    AP_GROUPINFO("GYR1", 7, AP_InertialSensor::TCal, gyro_coeff[0], 0),
+
+    // @Param: GYR2_X
+    // @DisplayName: Gyroscope 2nd order temperature coefficient X axis
+    // @Description: This is the 2nd order temperature coefficient from a temperature calibration
+    // @User: Advanced
+    // @Calibration: 1
+
+    // @Param: GYR2_Y
+    // @DisplayName: Gyroscope 2nd order temperature coefficient Y axis
+    // @Description: This is the 2nd order temperature coefficient from a temperature calibration
+    // @User: Advanced
+    // @Calibration: 1
+
+    // @Param: GYR2_Z
+    // @DisplayName: Gyroscope 2nd order temperature coefficient Z axis
+    // @Description: This is the 2nd order temperature coefficient from a temperature calibration
+    // @User: Advanced
+    // @Calibration: 1
+
+    AP_GROUPINFO("GYR2", 8, AP_InertialSensor::TCal, gyro_coeff[1], 0),
+
+    // @Param: GYR3_X
+    // @DisplayName: Gyroscope 3rd order temperature coefficient X axis
+    // @Description: This is the 3rd order temperature coefficient from a temperature calibration
+    // @User: Advanced
+    // @Calibration: 1
+
+    // @Param: GYR3_Y
+    // @DisplayName: Gyroscope 3rd order temperature coefficient Y axis
+    // @Description: This is the 3rd order temperature coefficient from a temperature calibration
+    // @User: Advanced
+    // @Calibration: 1
+
+    // @Param: GYR3_Z
+    // @DisplayName: Gyroscope 3rd order temperature coefficient Z axis
+    // @Description: This is the 3rd order temperature coefficient from a temperature calibration
+    // @User: Advanced
+    // @Calibration: 1
+
+    AP_GROUPINFO("GYR3", 9, AP_InertialSensor::TCal, gyro_coeff[2], 0),
+
+    AP_GROUPEND
+};
+
+
+/*
+  evaluate a 3rd order polynomial (without the constant term) given a set of coefficients
+ */
+Vector3f AP_InertialSensor::TCal::polynomial_eval(float tdiff, const AP_Vector3f coeff[3]) const
+{
+    // evaluate order 3 polynomial
+    const Vector3f *c = (Vector3f *)&coeff[0];
+    return (c[0] + (c[1] + c[2]*tdiff)*tdiff)*tdiff;
+}
+
+/*
+  correct a single sensor for the current temperature
+ */
+void AP_InertialSensor::TCal::correct_sensor(float temperature, float cal_temp, const AP_Vector3f coeff[3], Vector3f &v) const
```

Comment:
const AP_Vector3f&

Reply:
it is const ? can't do ref to array I think

Reply:
You are right